src/boot/pe.c

```diff
@@ -441,6 +441,13 @@ EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address, size_t
         if (!verify_pe(dos, pe, /* allow_compatibility= */ true))
                 return EFI_LOAD_ERROR;
 
+        if (ret_image_base) {
+                if (pe->OptionalHeader.Magic == OPTHDR32_MAGIC)
+                        *ret_image_base = pe->OptionalHeader.ImageBase32;
+                else
+                        *ret_image_base = pe->OptionalHeader.ImageBase64;
+        }
```

Comment:
so, hmm, this was broken before already, but we actually have this rule in systemd that functions either return success and then strictly initialize *all* return parameters, or they return failure and then do not touch any return parameters. this function doesn't follow this rule, and if you touch this please fix this.

or more specifically, we initialize ret_size_in_memory below, but then still have a chance to fail one line later.

the usual way to fix htis, is store all things to eventually return on success in a temporary variable first, and then copy things over only in the final success codepath where the function cannot fail anymore.